plugins/module_utils/network/sonic/facts/ntp/ntp.py

```diff
@@ -128,10 +128,12 @@ def get_ntp_configuration(self):
         servers = []
         for ntp_server in ntp_servers:
             if 'config' in ntp_server:
-                server = ntp_server['config']
-                if 'key-id' in server:
-                    server['key_id'] = server['key-id']
-                    server.pop('key-id')
+                server = {}
+                server['address'] = ntp_server['config'].get('address', None)
+                if 'key-id' in ntp_server['config']:
+                    server['key_id'] = ntp_server['config']['key-id']
```

Comment:
Please modify the logic here to only set the minpoll and maxpoll values into the config if the user has requested that values be set.

e.g.

```
if 'minpoll' in ntp_server['config']:
    minpoll = ntp_server['config'].get('minpoll', None)
    if minpoll is not None:
    server['minpoll'] = minpoll
```
This will avoid the possibility of putting "None" values into the config request.